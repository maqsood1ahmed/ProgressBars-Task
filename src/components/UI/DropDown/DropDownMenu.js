import React from 'react';
import { Select } from 'antd';

import propTypes from 'prop-types';

const DropDownMenu = props => {
    let { optionsList } = props;
    let DropDownOptionsJSX = optionsList.map((option, index) => {
        return (
            <Select.Option key={index} value={index}>
                {' '}
                #progress{index + 1}
            </Select.Option>
        );
    });
    return (
        <Select
            defaultValue={0}
            style={{ width: 120 }}
            onChange={e => props.handleChange(e)}
        >
            {DropDownOptionsJSX}
        </Select>
    );
};

DropDownMenu.propTypes = {
    optionsList: propTypes.array,
    handleChange: propTypes.func
};

DropDownMenu.defaultProps = {
    optionsList: []
};

export default DropDownMenu;
