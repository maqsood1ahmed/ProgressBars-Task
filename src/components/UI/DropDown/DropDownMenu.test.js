import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import DropDownMenu from './DropDownMenu';

configure({ adapter: new Adapter() });

describe('DropDownMenu test', () => {
    it('should be defined', () => {
        expect(DropDownMenu).toBeDefined();
    });
    it('should render correctly', () => {
        const tree = shallow(<DropDownMenu optionsList={[0, 1, 2]} />);
        expect(tree).toMatchSnapshot();
    });
});
