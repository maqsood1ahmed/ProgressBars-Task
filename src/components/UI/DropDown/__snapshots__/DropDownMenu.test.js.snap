// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDownMenu test should render correctly 1`] = `ShallowWrapper {}`;
